src/Orleans.Streaming.Redis/Providers/Streams/RedisQueueAdapterReceiver.cs

```diff
@@ -39,38 +54,56 @@ private RedisAdapterReceiver(ILogger logger, QueueId queueId, RedisDataManager q
             Id = queueId;
             _queue = queue ?? throw new ArgumentNullException(nameof(queue));
             _dataAdapter = dataAdapter;
-            _logger = logger.ForContext<RedisAdapterReceiver>();
+            _logger = logger != null ? logger.ForContext<RedisQueueAdapterReceiver>() : SilentLogger.Logger;
         }
 
-        public Task Initialize(TimeSpan timeout)
+        public async Task Initialize(TimeSpan timeout)
         {
-            if (_queue != null) // check in case we already shut it down.
+            using (var cts = new CancellationTokenSource(timeout))
             {
-                return _queue.InitAsync();
-            }
+                if (cts.IsCancellationRequested) throw new TaskCanceledException();
 
-            return Task.CompletedTask;
+                if (_queue != null) // check in case we already shut it down.
+                {
+                    await _queue.InitAsync(cts.Token);
+
+                    if (cts.IsCancellationRequested) throw new TaskCanceledException();
+
+                    await _queue.SubscribeAsync(cts.Token);
+
+                    if (cts.IsCancellationRequested) throw new TaskCanceledException();
```

Comment:
Is it typical to throw `TaskCanceledException` when the task has completed (since this falls after `SubscribeAsync`)?

Reply:
What's there is fine by me, particularly if tests are more predictable.